app-api/src/main/resources/logback-spring.xml

```diff
@@ -21,8 +21,23 @@
             <maxFileSize>${LOGS_MAX_FILE_SIZE}</maxFileSize>
         </rollingPolicy>
     </appender>
+    
+    <appender name="binlog-offset-errors" class="org.openmrs.eip.app.logger.BinlogReaderOffsetErrorsAppender">
```

Comment:
Do you really need a separate appender?

Reply:
We need to have a trigger to hook the debezium offset error occurence (which happens while the  DebeziumConsumer is getting up and running, during route debezium-route.xml loading) with their recovery mechanism. But it turned not to be as trivial as we thought, because, the debezium engine is asynchronously loaded, and, by default, the exception is only logged.

Below you can find the attempts:
1. Use the exceptionHandler in the debezium-route.xml uri as explained [here](https://camel.apache.org/components/3.18.x/debezium-mysql-component.html#_endpoint_query_option_exceptionHandler)
But, we found that this handler is only used to handle errors during message processing, after the DebeziumConsumer is successfully loaded.

2. Use a custom completionCallback
As we can see [here](https://github.com/debezium/debezium/blob/e705ba3a3dce54d2fbbf880a5270e62abbb4b61d/debezium-embedded/src/main/java/io/debezium/embedded/EmbeddedEngine.java#L962), the engine supports a callback to handle the shutdown of the engine (due to offset error, etc…). By default, this callback has only instructions to log the completion result, as we can see [here](https://github.com/debezium/debezium/blob/e705ba3a3dce54d2fbbf880a5270e62abbb4b61d/debezium-embedded/src/main/java/io/debezium/embedded/EmbeddedEngine.java#L610-L614).
We tried to send this callback as parameter in the DebeziumConsumer, but the method that builds the EmbeddedEngine is private, giving no space for extension/customization: see org.apache.camel.component.debezium.DebeziumConsumer.createDbzEngine(), lines 67-70.

So, we ended creating this async appender to listen to the debezium offset error, and trigger their recover mechanism. What do you think?

Reply:
Rather than doing this, isn't there a way to make the debezium engine to start synchronously? Alternatively I believe the camel component let's use a custom error handler that will fail and not ignore errors like the built-in one, see the `exceptionHandler` endpoint parameter, it will help avoid all this boilerpoint code

Reply:
I have just seen your first bullet, so the exception handler fails at start up, I still think there is a way, I will try to think about this.